lib/invoker.js

```diff
@@ -24,21 +26,64 @@ const invoke = (userFunction) => {
     };
 };
 
-module.exports = (userFunctionUri, port) => {
-    const userFunction = lookUpFunction(userFunctionUri);
+const registerShutdown = (destroyHook, timeout, server, shutdownSignal) => {
+    return shutdownSignal
+        .then(() => {
+            logger('Graceful shutdown started');
+            const gracefulShutdown = util.promisify(server.tryShutdown).bind(server);
+            return gracefulShutdown();
+        })
+        .then(
+            () => executeDestroy(destroyHook, timeout),
+            (shutdownErr) => {
+                logger(`Error during graceful shutdown: ${shutdownErr.toString()}`);
+                return executeDestroy(destroyHook, timeout).finally(() => Promise.reject(shutdownErr));
```

Comment:
finally returns a promise with the same resolution that it's chained from, not a promise for a new value. This is equivalent to a `try-catch-finally` statement with a return in the try/catch blocks, the finally will still execute, but it can't modify the returned value.

Reply:
oh snap :( undoing these, then